modules/server/src/main/ts/bedrock/server/EffectUtils.ts

```diff
@@ -1,71 +1,64 @@
-import { BrowserObject, Element } from 'webdriverio';
+import { Browser, Element } from 'webdriverio';
 
-export type ElementWithActions = Element & {
-  performActions: (actions: Array<Record<string, any>>) => Promise<void>;
-  releaseActions: () => Promise<void>;
+export type ElementWithActions = Element<'async'> & {
+  readonly performActions: (actions: Array<Record<string, any>>) => Promise<void>;
+  readonly releaseActions: () => Promise<void>;
 };
 
-const frameSelected = (driver: BrowserObject, frame: string): () => Promise<boolean> => {
-  return () => {
-    return driver.switchToFrame(frame).then(() => {
-      return true;
-    }).catch((e) => {
-      if (!(e.name && e.name === 'no such frame')) {
-        throw e;
-      }
-      return false;
-    });
-  };
+// eslint-disable-next-line @typescript-eslint/ban-types
+const frameSelected = (driver: Browser<'async'>, frame: object) => async (): Promise<boolean> => {
```

Comment:
Is it possible to be more specific with the type here, and indicate specifically that it needs to be the web driver element type (based on my quick scan of the docs)?

Reply:
I have no idea what you mean here sorry...

Reply:
Oh, are you talking above `object` type for `frame`? That's unchanged here so that'd really be out of scope.

Reply:
Yeah the `object` type. I think it was `string` previously?

Reply:
Fixed... I remember why I did this now,  it's because I was spiking and the type isn't exposed, as it's an `ElementReference`, not an `Element` as you mentioned.  As such, we need to get the type using `ReturnType`.